packages/js/test/plugins/auctionHouseModule/buy.test.ts

```diff
@@ -0,0 +1,68 @@
+import test, { Test } from 'tape';
+import spok, { Specifications } from 'spok';
+import { sol, token } from '@/types';
+import {
+  createNft,
+  createWallet,
+  killStuckProcess,
+  metaplex,
+  spokSameAmount,
+  spokSamePubkey,
+} from '../../helpers';
+import { createAuctionHouse } from './helpers';
+import { findAssociatedTokenAccountPda, Purchase } from '@/index';
+
+killStuckProcess();
+
+test('[auctionHouseModule] buy on an Auction House', async (t: Test) => {
+  // Given we have an Auction House and an NFT.
+  const mx = await metaplex();
+  const buyer = await createWallet(mx);
+
+  const nft = await createNft(mx);
+  const { auctionHouse } = await createAuctionHouse(mx);
+
+  // And we listed that NFT for 1 SOL.
+  const { listing } = await mx
+    .auctionHouse()
+    .list({
+      auctionHouse,
+      mintAccount: nft.address,
+      price: sol(1),
+    })
+    .run();
+
+  // When we execute direct buy with the given listing.
```

Comment:
We need more checks here on the state of every account. Buyer, Seller and Purchase.